torch/distributed/fsdp/_init_utils.py

```diff
@@ -586,42 +592,27 @@ def _get_ignored_modules(
 def _get_ignored_params(
     root_module: torch.nn.Module,
     ignored_modules: Set[torch.nn.Module],
-) -> Tuple[Set[torch.nn.Parameter], Set[str]]:
+    ignored_parameters: Optional[Iterable[torch.nn.Parameter]],
+) -> Set[torch.nn.Parameter]:
     """
-    Returns the parameters of the modules in ``ignored_modules``,
-    excluding any :class:`FlatParameter` s, and their fully prefixed names,
-    both as :class:`set` s.
+    Returns the parameters of the modules in ``ignored_modules`` and 
+    the parameters in ``ignored_parameters``, excluding any :class:`FlatParameter` s.
     """
-    ignored_params_to_names = dict(
-        (p, n)
+    all_ignored_params = set()
+    params_in_ignored_modules = list(
+        p
         for m in ignored_modules
-        for n, p in m.named_parameters()
+        for p in m.parameters()
         if not _is_fsdp_flattened(p)
     )
-    # Conservatively include all shared parameters' names
-    param_to_unflat_param_names = _get_param_to_fqns(
-        root_module,
-        dedup_shared_params=False,
+    params_in_ignored_parameters = list(
+        p
+        for p in ignored_parameters
+        if not _is_fsdp_flattened(p)
     )
-    ignored_param_names = set()
-    for param, name in ignored_params_to_names.items():
-        if param not in param_to_unflat_param_names:
-            # Allow users to pass parameters not under FSDP root module.
-            # This is useful when user apply FSDP manually to different
-            # submodules with the same global set of ignored parameters.
-            warnings.warn(
-                f"Parameter {name} is in the ignored modules passed to FSDP, "
-                "but it's not under the root module wrapped by FSDP."
-            )
-            continue
-
-        unflat_param_names = param_to_unflat_param_names[param]
-        clean_names = []
-        for k in unflat_param_names:
-            # Clean any module wrapper prefixes in case of nested wrapping
-            clean_names.append(clean_tensor_name(k))
-        ignored_param_names.update(clean_names)
-    return set(ignored_params_to_names.keys()), ignored_param_names
+    all_ignored_params.update(params_in_ignored_modules, params_in_ignored_parameters)
```

Comment:
in theory exactly only one of these (or neither I guess) should be populated right, since we only allow passing one or the other for now?

Reply:
yes, the unit tests have already covered this case